mmhuman3d/data/data_structures/smc_reader.py

```diff
@@ -519,3 +533,153 @@ def get_depth_floor(self, device_id: int) -> dict:
             return device_dict['floor']
         else:
             raise KeyError(f'Kinect {device_id} has no floor data.')
+
+    def _get_keypoints2d(self, device, device_id, frame_id):
+        """Get keypoints2d projected from keypoints3d.
+
+        Args:
+            device (str):
+                Device name, should be Kinect or iPhone.
+            device_id (int):
+                ID of a device, starts from 0.
+            frame_id (int, list or None, optional):
+                int: frame id of one selected frame
+                list: a list of frame id
+                None: all frames will be returned
+                Defaults to None.
+
+        Returns:
+            Tuple[np.ndarray, np.ndarray]: keypoints2d and its mask
+        """
+        assert device in {'Kinect', 'iPhone'}
+        kps2d_dict = self.smc['Keypoints2D'][device][str(device_id)]
+        keypoints2d = kps2d_dict['keypoints2d'][...]
+        keypoints2d_mask = kps2d_dict['keypoints2d_mask'][...]
+
+        if frame_id is not None:
+            if isinstance(frame_id, int):
+                frame_id = [frame_id]
+            keypoints2d = keypoints2d[frame_id, ...]
+        return keypoints2d, keypoints2d_mask
+
+    def get_kinect_keypoints2d(self, device_id, frame_id=None):
+        assert device_id < self.num_kinects
```

Comment:
and device_id >= 0 ?

Reply:
Added in https://github.com/open-mmlab/mmhuman3d/pull/66/commits/b2c2b53d942e602db0427c184427a7371a18704e